pkgs/development/python-modules/Theano/default.nix

```diff
@@ -81,7 +80,8 @@ in buildPythonPackage rec {
 
   # keep Nose around since running the tests by hand is possible from Python or bash
   checkInputs = [ nose ];
-  propagatedBuildInputs = [ numpy numpy.blas scipy six libgpuarray_ ];
+  propagatedBuildInputs = [ numpy numpy.blas scipy six libgpuarray_ ]
+    ++ lib.optionals cudaSupport [ setuptools ];
```

Comment:
```suggestion
  propagatedBuildInputs = [ numpy numpy.blas setuptools scipy six libgpuarray_ ];
```

Reply:
I was as surprised as you but this really is needed on cuda. Without it cuda-specific submodules fail to import.

Reply:
I think we can cheat a bit and always add it to prevent bugs in the future. setuptools is very common and it is easy to miss because pythonImportsCheck impurely has it in its environment which means we can't easily detect when it is missing.

Reply:
Okay, added.